File: themes/theme-001.py
# Read content from HTML files
with open("marksheet.html", "r") as file:
    marksheet = file.read()

with open("_header.html", "r") as file:
    header = file.read()

with open("_scores.html", "r") as file:
    scores = file.read()

with open("_summary.html", "r") as file:
    summary = file.read()

with open("_gradings.html", "r") as file:
    gradings = file.read()

# Replace placeholders with content
theme = marksheet
theme = theme.replace("<!-- #header-html -->", header)
theme = theme.replace("<!-- #scores-table -->", scores)
theme = theme.replace("<!-- #summary-table -->", summary)
theme = theme.replace("<!-- #gradings-table -->", gradings)

# Write the final output to a new file
with open("theme-000.html", "w") as file:
    file.write(theme)

File: themes/theme-001.php
<?php
$marksheet = file_get_contents("marksheet.html");
$header = file_get_contents("_header.html");
$scores = file_get_contents("_scores.html");
$summary = file_get_contents("_summary.html");
$gradings = file_get_contents("_gradings.html");

$css = '<link rel="stylesheet" href="marksheet.css" />';
$theme = $marksheet;
$theme = str_replace("<!-- #header-html -->", $header, $theme);
$theme = str_replace("<!-- #scores-table -->", $scores, $theme);
$theme = str_replace("<!-- #summary-table -->", $summary, $theme);
$theme = str_replace("<!-- #gradings-table -->", $gradings, $theme);
$theme = str_replace($css, '', $theme);
$theme = str_replace('<!-- #css -->', $css, $theme);

file_put_contents("theme-000.html", $theme);
